api/intel/openamt/openamt.go

```diff
@@ -0,0 +1,152 @@
+package openamt
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+const (
+	MpsServerAddress          = "127.0.0.1" // TODO: this is the same IP that must be used for the OpenAMT stack deployment, is this value right?
```

Comment:
its very much not able to be a loopback address - as each container (including the Portainer one) has its own.
I _suspect_ we're going to need to detect which stack is running, and look into the MPS container config

but even then, that assumes that the MPS container is on the same node as either portainer - and really, it could be running on some other environment. (lets assume for the first cut, that there's only one openAMT stack running, and its running on the same node as the portainer backend, and that backend is Docker based - .... I wonder if its easier for the "enable openAMT" to request the stack is started, or for it to detect the stack, and ask for the IP address.

Reply:
is this the `MPS_COMMON_NAME` setting in the app template?

I set mine to `p1` (the hostname of my notebook), and then can talk to it using `https://p1:4433` - tho with a self signed cert, so that might be "fun" too

Reply:
Yep that value should be the same as `MPS_COMMON_NAME`, I'll change it so it is submitted by the user (as well as the password), same as the app template form